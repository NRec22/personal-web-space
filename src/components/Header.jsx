import React from 'react';
import Navbar from 'react-bulma-components/lib/components/navbar';
import Container from 'react-bulma-components/lib/components/container';

const Header = () => (
  <Navbar>
    <Container>
      <Navbar.Brand>
        <Navbar.Item>...</Navbar.Item>
        <Navbar.Burger />
      </Navbar.Brand>
      <Navbar.Menu>
        <Navbar.Container>
          <Navbar.Item>Home</Navbar.Item>
          <Navbar.Item>About Me</Navbar.Item>
          <Navbar.Item>LinkedIn</Navbar.Item>
          <Navbar.Item>Github</Navbar.Item>
          <Navbar.Item>Contact Me</Navbar.Item>
        </Navbar.Container>
      </Navbar.Menu>
    </Container>
  </Navbar>
);

export default Header;