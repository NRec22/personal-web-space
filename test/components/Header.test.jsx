import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Navbar from 'react-bulma-components/lib/components/navbar';
import Header from '../../src/components/Header';

describe('Header: ', () => {
    it('contains a navbar', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find(Navbar)).toBeDefined;
    })

    it('contains navigation titles', () => {
        const wrapper = shallow(<Header />);
        const navText = ['Home', 'About Me', 'Contact', 'LinkedIn', 'Github'];

        navText.forEach((text) => {
            console.info(wrapper.find(Navbar).text());
            expect(wrapper.text()).toContain(text);
        });
    })

    it('should render as expected', () => {
        const wrapper = shallow(<Header />);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    })
})
