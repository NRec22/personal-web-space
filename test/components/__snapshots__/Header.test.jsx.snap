// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header:  should render as expected 1`] = `
<Navbar
  className=""
  color={null}
  fixed={null}
  renderAs="nav"
  style={Object {}}
  transparent={false}
>
  <Container
    breakpoint={null}
    className=""
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Navbar.Brand
      active={false}
      className=""
      style={Object {}}
      toggleMenu={[Function]}
    >
      <Navbar.Item
        active={false}
        className=""
        dropdown={false}
        dropdownUp={false}
        hoverable={false}
        renderAs="a"
        style={Object {}}
      >
        ...
      </Navbar.Item>
      <Navbar.Burger
        active={false}
        className=""
        onClick={[Function]}
        style={Object {}}
        toggleMenu={[Function]}
      />
    </Navbar.Brand>
    <Navbar.Menu
      active={false}
      className=""
      style={Object {}}
    >
      <Navbar.Container
        className=""
        position="start"
        renderAs="div"
        style={Object {}}
      >
        <Navbar.Item
          active={false}
          className=""
          dropdown={false}
          dropdownUp={false}
          hoverable={false}
          renderAs="a"
          style={Object {}}
        >
          Home
        </Navbar.Item>
        <Navbar.Item
          active={false}
          className=""
          dropdown={false}
          dropdownUp={false}
          hoverable={false}
          renderAs="a"
          style={Object {}}
        >
          About Me
        </Navbar.Item>
        <Navbar.Item
          active={false}
          className=""
          dropdown={false}
          dropdownUp={false}
          hoverable={false}
          renderAs="a"
          style={Object {}}
        >
          LinkedIn
        </Navbar.Item>
        <Navbar.Item
          active={false}
          className=""
          dropdown={false}
          dropdownUp={false}
          hoverable={false}
          renderAs="a"
          style={Object {}}
        >
          Github
        </Navbar.Item>
        <Navbar.Item
          active={false}
          className=""
          dropdown={false}
          dropdownUp={false}
          hoverable={false}
          renderAs="a"
          style={Object {}}
        >
          Contact Me
        </Navbar.Item>
      </Navbar.Container>
    </Navbar.Menu>
  </Container>
</Navbar>
`;
